tests/test_forward.cpp

```diff
@@ -0,0 +1,246 @@
+#include <gtest/gtest.h>
+#include "datapath\forwarding.h"
+#include "datapath\signals.h"
+
+TEST(Forward, detect_EXMEM_hazard_ALUSRC1)
+{
+	Forward fwd;
+	FRWD_Out fwd_out;
+	IDEX idex;
+	EXMEM exmem;
+	MEMWB memwb;
+	
+	//ID/EX
+	idex.read1 = 0x0002;
+	idex.read2 = 0x0004;
+	
+	//EX/ME
+	exmem.write_reg = 0x0002;
+
+	//MEM/WB
+	memwb.wb_controls.mem_to_reg = 0;
+	memwb.write_reg = 0x1234;
```

Comment:
This does not mean what you think it means.  `write_reg` holds the register number that will be written, not the data that will be written to the register.